tools/autograd/derivatives.yaml

```diff
@@ -177,14 +177,14 @@
   batch2: batch1.transpose(1, 2).bmm(grad.unsqueeze(0).expand({ batch1.size(0), batch1.size(1), batch2.size(2) })) * alpha
 
 - name: addcdiv(Tensor self, Tensor tensor1, Tensor tensor2, *, Scalar value=1) -> Tensor
-  self: grad
-  tensor1: grad * value / tensor2
-  tensor2: -grad * value * tensor1 / (tensor2 * tensor2)
+  self: handle_r_to_c(self.scalar_type(), grad)
+  tensor1: handle_r_to_c(tensor1.scalar_type(), grad * (value / tensor2).conj())
+  tensor2: handle_r_to_c(tensor2.scalar_type(), -grad * (value * tensor1 / (tensor2 * tensor2)).conj())
 
 - name: addcmul(Tensor self, Tensor tensor1, Tensor tensor2, *, Scalar value=1) -> Tensor
-  self: grad
-  tensor1: grad * tensor2 * value
-  tensor2: grad * tensor1 * value
+  self: handle_r_to_c(self.scalar_type(), grad)
```

Comment:
`addcmul` and `addcdiv` currently don't support type promotion rn so `R -> C` case would never occur but adding this as a precaution in case the type promotion is enabled for these ops in future.